networkx/algorithms/community/tests/test_quality.py

```diff
@@ -73,6 +74,64 @@ def test_modularity():
     assert almost_equal(2 / 9, modularity(G, C))
 
 
+def test_modularity_resolution():
+    G = nx.barbell_graph(3, 0)
+    C = [{0, 1, 4}, {2, 3, 5}]
+    assert modularity(G, C) == pytest.approx(3 / 7 - 100 / 14 ** 2)
+    gamma = 2
+    result = modularity(G, C, resolution=gamma)
+    assert result == pytest.approx(3 / 7 - gamma * 100 / 14 ** 2)
+    gamma = 0.2
+    result = modularity(G, C, resolution=gamma)
+    assert result == pytest.approx(3 / 7 - gamma * 100 / 14 ** 2)
+
+    C = [{0, 1, 2}, {3, 4, 5}]
+    assert modularity(G, C) == pytest.approx(6 / 7 - 98 / 14 ** 2)
+    gamma = 2
+    result = modularity(G, C, resolution=gamma)
+    assert result == pytest.approx(6 / 7 - gamma * 98 / 14 ** 2)
+    gamma = 0.2
+    result = modularity(G, C, resolution=gamma)
+    assert result == pytest.approx(6 / 7 - gamma * 98 / 14 ** 2)
+
+    G = nx.barbell_graph(5, 3)
+    C = [frozenset(range(5)), frozenset(range(8, 13)), frozenset(range(5, 8))]
+    gamma = 1
+    result = modularity(G, C, resolution=gamma)
+    assert result == pytest.approx((22 / 24) - gamma * (918 / (48 ** 2)))
+    assert result == pytest.approx(0.518229)  # This C is maximal for gamma=1
```

Comment:
```suggestion
    assert result == pytest.approx((22 / 24) - gamma * (918 / (48 ** 2)))
```
The floating point value 0.518229 is just the evaluation of the expression on the line above correct? If so, then these asserts are redundant.

Note: ditto the next two comments.